project/src/test/resources/docker/sql/create-schema.sql

```diff
@@ -469,3 +475,223 @@ CREATE TABLE dbo.policies(
 	version VARCHAR(10),	--	v0, v1...
 	path VARCHAR(100)		-- .../courses
 );
+
+-----------------------------------------------------
+-- Write API
+
+CREATE OR REPLACE PROCEDURE dbo.sp_createOrReplaceSchool (
+  schoolName VARCHAR(100),
+  schoolAcr VARCHAR(50),
+  programmeName VARCHAR(100),
+  programmeAcr VARCHAR(50),
+  programmeTermSize INT,
+  calendarTerm VARCHAR(50))
+AS $$
+#print_strict_params ON
+BEGIN
+  IF (schoolName IS NULL AND schoolAcr IS NULL) OR (programmeAcr IS NULL AND programmeName IS NULL) THEN
+    RAISE 'For the School and Programme parameters, you must provide at least a "name" or an "acronym"';
+  END IF;
+
+  -- Insert or update Programme
+  -- coalesce: if any of the supplied parameters is null, use the previous value of the column instead
+  UPDATE
+    dbo.programme
+  SET
+    name     = COALESCE(programmeName, name),
+    acronym  = COALESCE(programmeAcr, acronym),
+    termSize = COALESCE(programmeTermSize, termSize)
+  WHERE
+    name = programmeName OR acronym = programmeAcr;
+
+  INSERT INTO
+    dbo.Programme(name, acronym, termSize)
+  SELECT
+    programmeName, programmeAcr, programmeTermSize
+  WHERE
+    NOT EXISTS(
+      SELECT id FROM dbo.Programme p
+      WHERE p.name = programmeName OR p.acronym=programmeAcr
+    );
+
+  -- Insert or update CalendarTerm
+  INSERT INTO
+    dbo.CalendarTerm(id, start_date, end_date)
+  SELECT
+    calendarTerm, NULL, NULL
+  WHERE
+    NOT EXISTS(
+      SELECT id FROM dbo.CalendarTerm c WHERE c.id=calendarTerm
+    ) AND calendarTerm IS NOT NULL;
+
+END
+$$ LANGUAGE plpgsql;
+
+CREATE OR REPLACE PROCEDURE dbo.sp_createOrReplaceCourse (
+  courseName VARCHAR(100),
+  courseAcr VARCHAR(50),
+  calendarSection VARCHAR(50),
+  calTerm VARCHAR(50),
+  lang VARCHAR(10),
+	categ VARCHAR(100))
+AS $$
+#print_strict_params ON
+DECLARE
+	clid INT;
+	cid INT;
+	cscal INT;
+  langid INT;
+  categoryid INT;
+BEGIN
+  IF (courseName IS NULL AND courseAcr IS NULL) THEN
+    RAISE 'For the Course parameters, you must provide at least a "name" or an "acronym"';
+  END IF;
+
+  SELECT id INTO langid FROM dbo.Language WHERE name = lang;
+  IF (langid IS NULL) THEN
+    RAISE '% language does not exist', lang;
+  END IF;
+
+  SELECT
+    c.id INTO categoryid
+  FROM
+    dbo.category c JOIN
+    dbo.categorylanguage cl ON c.id = cl.category
+  WHERE
+    name = categ AND language = langid;
+  IF (categoryid IS NULL) THEN
+    RAISE '"%" is not a valid category for the language %', categ, lang;
+  END IF;
```

Comment:
Is this for checking if a category with the course name exists or what kind of category should this be?

Reply:
This is checking if the category of the event specified in the HTTP request is valid. Since categories are directly inserted to the database and never modified by the write API.

Reply:
Does the Integration project choose the categories of the event then? Do they know which ones exist currently?

Reply:
We decided to use the Category ID (Integer) and list the available IDs in the Write API documentation.